src/main/java/tr/com/srdc/cda2fhir/transform/CCDTransformerImpl.java

```diff
@@ -92,122 +92,189 @@ public CCDTransformerImpl() {
+	}
+
+	public void setIdGenerator(IdGeneratorEnum idGen) {
+		this.idGenerator = idGen;
+	}
+
+	public void addSection(CDASectionTypeEnum sectionEnum) {
+		supportedSectionTypes.add(sectionEnum);
+	}
+
+	/**
+	 * @param cda                A Consolidated CDA (C-CDA) 2.1 Continuity of Care
+	 *                           Document (CCD) instance to be transformed
+	 * @param bundleType         Desired type of the FHIR Bundle to be returned
+	 * @param resourceProfileMap The mappings of default resource profiles to
+	 *                           desired resource profiles. Used to set profile
+	 *                           URI's of bundle entries or omit unwanted entries.
+	 * @return A FHIR Bundle that contains a Composition corresponding to the CCD
+	 *         document and all other resources but Patient that are referenced
+	 *         within the Composition.
+	 */
+	public Bundle createTransactionBundle(Bundle bundle, Map<String, String> resourceProfileMap, boolean addURLs) {
+		Bundle resultBundle = new Bundle();
+
+		for (BundleEntryComponent entry : bundle.getEntry()) {
+			// Patient resource will not be added
+			if (entry != null) {
+				// Add request and fullUrl fields to entries
+				addRequestToEntry(entry);
+				if (addURLs) {
+					addFullUrlToEntry(entry);
+				}
+				// if resourceProfileMap is specified omit the resources with no profiles given
+				// Empty profileUri means add with no change
+				if (resourceProfileMap != null) {
+					String profileUri = resourceProfileMap.get(entry.getResource().getResourceType().name());
+					if (profileUri != null) {
+						if (!profileUri.isEmpty()) {
+							entry.getResource().getMeta().addProfile(profileUri);
+						}
+						resultBundle.addEntry(entry);
+					}
+				} else {
+					resultBundle.addEntry(entry);
+				}
+			}
+		}
+
+		return resultBundle;
+	}
+
+	/**
+	 * Transforms a Consolidated CDA (C-CDA) 2.1 Continuity of Care Document (CCD)
+	 * instance to a Bundle of corresponding FHIR resources
+	 * 
+	 * @param cda                A Consolidated CDA (C-CDA) 2.1 Continuity of Care
+	 *                           Document (CCD) instance to be transformed
+	 * @param bundleType         The type of bundle to create, currently only
+	 *                           supports transaction bundles.
+	 * @param resourceProfileMap The mappings of default resource profiles to
+	 *                           desired resource profiles. Used to set profile
+	 *                           URI's of bundle entries or omit unwanted entries.
+	 * @param documentBody       The decoded documentBody of the document, to be
+	 *                           included in a provenance object.
+	 * @return A FHIR Bundle that contains a Composition corresponding to the CCD
+	 *         document and all other resources that are referenced within the
+	 *         Composition.
+	 * @throws Exception
+	 */
+	public Bundle transformDocument(String filePath, BundleType bundleType, Map<String, String> resourceProfileMap,
```

Comment:
So this one is a little weird... since we are feeding in a filePath, we don't really need to pass in documentBody. That would mean we'd need to read in the document before passing it to CDA2FHIR, then let CDA2FHIR read it again which is very inefficent. However, this is more of an API design problem and not an implementing provenance problem. We need to revisit all of these, so I ticketed it out and think this is fine for now.